manifests/mod/jk.pp

```diff
@@ -0,0 +1,105 @@
+# Class apache::mod::jk
+#
+# Manages mod_jk connector
+#
+# All parameters are optional. When undefined, some receive default values,
+# while others cause an optional directive to be absent
+#
+# For help on parameters, pls see official reference at:
+# https://tomcat.apache.org/connectors-doc/reference/apache.html
+#
+class apache::mod::jk (
+  $workers_file          = undef,
+  $worker_property       = {},
+  $shm_file              = "${::apache::logroot}/jk-runtime-status",
+  $shm_size              = undef,
+  $mount_file            = undef,
+  $mount_file_reload     = undef,
+  $mount                 = {},
+  $un_mount              = {},
+  $auto_alias            = undef,
+  $mount_copy            = undef,
+  $worker_indicator      = undef,
+  $watchdog_interval     = undef,
+  $log_file              = "${::apache::logroot}/mod_jk.log",
+  $log_level             = undef,
+  $log_stamp_format      = undef,
+  $request_log_format    = undef,
+  $extract_ssl           = undef,
+  $https_indicator       = undef,
+  $sslprotocol_indicator = undef,
+  $certs_indicator       = undef,
+  $cipher_indicator      = undef,
+  $certchain_prefix      = undef,
+  $session_indicator     = undef,
+  $keysize_indicator     = undef,
+  $local_name_indicator  = undef,
+  $ignore_cl_indicator   = undef,
+  $local_addr_indicator  = undef,
+  $local_port_indicator  = undef,
+  $remote_host_indicator = undef,
+  $remote_addr_indicator = undef,
+  $remote_port_indicator = undef,
+  $remote_user_indicator = undef,
+  $auth_type_indicator   = undef,
+  $options               = [],
+  $env_var               = {},
+  $strip_session         = undef,
+  # Location list
+  # See comments in template mod/jk.conf.erb
+  $location_list         = [],
+  # Workers file content
+  # See comments in template mod/jk/workers.properties.erb
+  $workers_file_content  = {},
+  # Mount file content
+  # See comments in template mod/jk/uriworkermap.properties.erb
+  $mount_file_content    = {},
+){
```

Comment:
What I exactly need is the mod/jk spec test to access the variables from ::apache class. If I understood you correctly, inheritance would make sure the variables get included before needed in the spec test. Is that correct?

I'm avoiding inheritance here because it's considered an anti-pattern by Puppet itself - see block "Aside: When to Inherit" [here](https://docs.puppet.com/puppet/5.0/lang_classes.html#inheritance).

An alternative would be to declare the parameter as undef in the class signature, then assingn values depending on such parameter, like I did with workers_file and mount_file. So, by assigning something in the test in :parameters section, maybe the required variables will be there. Doesn't seem much clear (I'm in a hurry now), but feel free to ask.

Thanks!